api/package.json

```diff
@@ -38,5 +39,22 @@
     "ts-loader": "^3.2.0",
     "typescript": "^2.6.2",
     "webpack": "^3.10.0"
+  },
+  "devDependencies": {
+    "@types/jest": "^22.0.1",
+    "jest": "^22.0.4"
+  },
+  "jest": {
+    "moduleFileExtensions": [
+      "ts",
+      "tsx",
+      "js"
+    ],
+    "transform": {
+      "^.+\\.(ts|tsx)$": "<rootDir>/jest-ts-preprocessor.js"
+    },
+    "testMatch": [
+      "**/__tests__/*.(ts|tsx|js)"
```

Comment:
We specified the `test.js` extension on the ui side because we ended up needed helpers for our tests. I'm not sure if it's worth prepping for that now.

Unrelated: do we want to allow JS or should everything be type script?

Reply:
Oh I like the idea of making it easy to add helpers to our tests if we need them, without having to modify the test configuration and rename tests.  Also, I like the idea of keeping the test naming conventions as similar as possible between the ui and api sides.

I think we should allow JS for now since we don't have a ton of institutional knowledge of TS at present, and should allow for the possibility that this TypeScript thing might explode in our face, or that it might just end up being more convenient in some cases to not use it. I think our code review process should be enough to ensure that we nudge each other towards TS when possible, without explicitly prohibiting JS in the build pipeline... let me know if you disagree though!